test/e2e/main/main_test.go

```diff
@@ -817,3 +818,66 @@ func throwHTTPError(resp *http.Response) {
 	errorBytes, _ := ioutil.ReadAll(resp.Body)
 	panic(fmt.Errorf("unexpected HTTP status code (%d): %s", resp.StatusCode, string(errorBytes)))
 }
+
+func TestCacheCR(t *testing.T) {
+	t.Parallel()
+	spec := tutils.DefaultSpec(testKube)
+	name := strcase.ToKebab(t.Name())
+	spec.Name = name
+	testKube.CreateInfinispan(spec, tutils.Namespace)
+	defer testKube.DeleteInfinispan(spec, tutils.SinglePodTimeout)
+	testKube.WaitForInfinispanPods(1, tutils.SinglePodTimeout, spec.Name, tutils.Namespace)
+	testKube.WaitForInfinispanCondition(spec.Name, spec.Namespace, ispnv1.ConditionWellFormed)
+
+	ispn := ispnv1.Infinispan{}
+	tutils.ExpectNoError(testKube.Kubernetes.Client.Get(context.TODO(), types.NamespacedName{Namespace: spec.Namespace, Name: spec.Name}, &ispn))
+
+	//Test for CacheCR with Templatename
+
+	cacheCRTemplateName := createCacheWithCR("cache-with-static-template", spec.Namespace, name)
+	cacheCRTemplateName.Spec.TemplateName = "org.infinispan.DIST_SYNC"
+	testCacheWithCR(&ispn, cacheCRTemplateName)
+
+	//Test for CacheCR with TemplateXML
+
+	cacheCRTemplateXML := createCacheWithCR("cache-with-xml-template", spec.Namespace, name)
+	cacheCRTemplateXML.Spec.Template = "<infinispan><cache-container><distributed-cache name=\"cache-with-xml-template\" mode=\"SYNC\"><persistence><file-store/></persistence></distributed-cache></cache-container></infinispan>"
+	testCacheWithCR(&ispn, cacheCRTemplateXML)
+}
+
+func testCacheWithCR(ispn *ispnv1.Infinispan, cache *v2alpha1.Cache) {
+	key := "testkey"
+	value := "test-operator"
+	testKube.Create(cache)
+	hostAddr, client := tutils.HTTPClientAndHost(ispn, testKube)
+	waitForCacheToBeCreated(cache.Spec.Name, hostAddr, client)
+	condition := v2alpha1.CacheCondition{
+		Type:   "Ready",
+		Status: "True",
+	}
+	testKube.WaitForCacheCondition(cache.Spec.Name, cache.ObjectMeta.Namespace, condition)
```

Comment:
We should wait for the Cache CR reconcilation to be ready before cache validation via REST API, i.e. `WaitForCacheCondition` should be before `waitForCacheToBeCreated`

Reply:
Pushed the changes